id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java

```diff
@@ -346,24 +333,30 @@ public DataProviderResponse getFormattedCredentialData(
 		return bestFingerList;
 	}
 
-	private String getSubType(List<String> bdbSubTypeList) {
-		String subType;
-		if (bdbSubTypeList.size() == 1) {
-			subType = bdbSubTypeList.get(0);
-		} else {
-			subType = bdbSubTypeList.get(0) + " " + bdbSubTypeList.get(1);
+	protected String getSubType(List<String> bdbSubTypeList) {
+		try {
+			String subType;
+			if (bdbSubTypeList.size() == 1) {
+				subType = bdbSubTypeList.get(0);
+			} else {
+				subType = bdbSubTypeList.get(0) + " " + bdbSubTypeList.get(1);
+			}
+			return subType;
+		} catch (Exception e) {
+			return null;

```

Comment:
Catch a specific exception. Add logger to the exception, if needed in debug level.

Reply:
Fixed